aten/src/ATen/native/LinearAlgebra.cpp

```diff
@@ -1414,6 +1414,16 @@ static void addmm_impl_cpu_(
   AT_DISPATCH_ALL_TYPES_AND_COMPLEX_AND(kBFloat16,
       result.scalar_type(), "addmm_impl_cpu_",
       [&]{
+#if defined(__aarch64__) && AT_MKLDNN_ACL_ENABLED()
+        // On AArch64 if LHS matrix in BLAS routine is transposed but RHS is not then
+        // it is faster to call oneDNN matrix multiplication primitive with RHS*LHS
+        // that will call then into ACL GEMM kernel and also additionally have support
+        // for running kernel with BF16 instructions
+        if(transpose_a && !transpose_b && result.scalar_type() == at::ScalarType::Float) {
```

Comment:
this needs to be guarded with bf16 check, as shown below, because mkldnn matmul is supported for only bf16 devices. otherwise this is going to break aarch64 non-bf16 devices.

`if(use_mkldnn_bf16_matmul(b, a, c) && transpose_a && !transpose_b && result.scalar_type() == at::ScalarType::Float) {`

Reply:
Hmm, not really, `mkldnn_matmul` supports fp32 too. We guarded it for bf16 but we didn't find performance benefit of it compared to MKL sgemm on x86 CPU. Things could be different for aarch64?

Reply:
Thanks @snadampal and @jgong5.

I have updated PR with [323af68](https://github.com/pytorch/pytorch/pull/91763/commits/323af68e24fc61520d4441f4f29de42b1255d2ff) that enables calling to `mkldnn_matmul` on AArch64 as in our tests we noticed performance improvements.

Reply:
Hi @jgong5, what I meant was the earlier version of the change as such would fail a check in mkldnn/Matmul.cpp because here we were allowing the devices with only bf16 support, I think it was for performance reasons on x86.

Hi @milpuz01 , thanks for fixing it.